src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/PartialRemissionHWFCallbackHandlerTest.java

```diff
@@ -0,0 +1,149 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CallbackType;
+import uk.gov.hmcts.reform.civil.enums.FeeType;
+import uk.gov.hmcts.reform.civil.handler.callback.BaseCallbackHandlerTest;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.Fee;
+import uk.gov.hmcts.reform.civil.model.citizenui.HelpWithFeesDetails;
+
+import java.math.BigDecimal;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.PARTIAL_REMISSION_HWF_GRANTED;
+import static uk.gov.hmcts.reform.civil.handler.callback.user.PartialRemissionHWFCallbackHandler.ERR_MSG_FEE_TYPE_NOT_CONFIGURED;
+import static uk.gov.hmcts.reform.civil.handler.callback.user.PartialRemissionHWFCallbackHandler.ERR_MSG_REMISSION_AMOUNT_LESS_THAN_CLAIM_FEE;
+import static uk.gov.hmcts.reform.civil.handler.callback.user.PartialRemissionHWFCallbackHandler.ERR_MSG_REMISSION_AMOUNT_LESS_THAN_HEARING_FEE;
+import static uk.gov.hmcts.reform.civil.handler.callback.user.PartialRemissionHWFCallbackHandler.ERR_MSG_REMISSION_AMOUNT_LESS_THAN_ZERO;
+
+@ExtendWith(MockitoExtension.class)
+public class PartialRemissionHWFCallbackHandlerTest extends BaseCallbackHandlerTest {
+
+    private PartialRemissionHWFCallbackHandler handler;
+
+    @BeforeEach
+    void setUp() {
+        var objectMapper = new ObjectMapper();
+        handler = new PartialRemissionHWFCallbackHandler(objectMapper);
+    }
+
+    @Test
+    void handleEventsReturnsTheExpectedCallbackEvent() {
+        assertThat(handler.handledEvents()).contains(PARTIAL_REMISSION_HWF_GRANTED);
+    }
+
+    @Nested
+    class AboutToSubmitCallback {
+
+        @Test
+        void shouldCallPartialRemissionHwfEvent() {
+            CaseData caseData = CaseData.builder()
+                .claimFee(Fee.builder().calculatedAmountInPence(BigDecimal.valueOf(10000)).code("OOOCM002").build())
+                .claimIssuedHwfDetails(HelpWithFeesDetails.builder()
+                                           .remissionAmount(BigDecimal.valueOf(1000))
+                                           .build())
+                .hwfFeeType(FeeType.CLAIMISSUED)
+                .build();
+
+            CallbackParams params = callbackParamsOf(caseData, CallbackType.ABOUT_TO_SUBMIT);
+
+            //When
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            //Then
+            assertThat(response.getErrors()).isNull();
+            assertThat(response.getData()).isNotNull();
```

Comment:
it would be better to compare the values with the object

Reply:
done